pkg/cluster/cluster.go

```diff
@@ -206,3 +212,73 @@ func GetClusterAddOns(connection *sdk.Connection, clusterID string) ([]*AddOnIte
 
 	return clusterAddOns, nil
 }
+
+// DoSurvey will ask user to choose one if there are more than one clusters match the query
+func DoSurvey(clusters []*clustersmgmtv1.Cluster) (cluster *clustersmgmtv1.Cluster, err error) {
+	clusterList := []string{}
+	for _, v := range clusters {
+		clusterList = append(clusterList, fmt.Sprintf("Name: %s, ID: %s", v.Name(), v.ID()))
+	}
+	choice := ""
+	prompt := &survey.Select{
+		Message: "Please choose a cluster:",
+		Options: clusterList,
+		Default: clusterList[0],
+	}
+	survey.PageSize = ClustersPageSize
+	err = survey.AskOne(prompt, &choice, func(ans interface{}) error {
+		choice := ans.(string)
+		found := false
+		for _, v := range clusters {
+			if strings.Contains(choice, v.ID()) {
+				found = true
+				cluster = v
+			}
+		}
+		if !found {
+			return fmt.Errorf("the cluster you choose is not valid: %s", choice)
+		}
+		return nil
+	})
+	return cluster, err
+}
+
+// FindClusters finds the clusters that match the given key. A cluster matches the key if its
+// identifier is that key, or if its name starts with that key. For example, the key `prd-2305`
+// doesn't match a cluster directly because it isn't a valid identifier, but it matches all clusters
+// whose names start with `prd-2305`.
+func FindClusters(collection *clustersmgmtv1.ClustersClient, key string,
```

Comment:
I less prefer using this function for finding clusters. It returns a list of clusters that match a key while in my opinion a more smooth experience will be to return just one cluster based on a name or ID.